crates/ark-contracts/arkchain/src/orderbook.cairo

```diff
@@ -291,32 +292,27 @@ mod orderbook {
         fn create_order(ref self: ContractState, order: OrderV1, signer: Signer) {
             let order_hash = order.compute_order_hash();
             let user_pubkey = SignerValidator::verify(order_hash, signer);
-
             let block_ts = starknet::get_block_timestamp();
             let validation = order.validate_common_data(block_ts);
             if validation.is_err() {
                 panic_with_felt252(validation.unwrap_err().into());
             }
-
             let order_type = order
                 .validate_order_type()
                 .expect(orderbook_errors::ORDER_INVALID_DATA);
-
             let order_hash = order.compute_order_hash();
-
-            // Check if order already exists
             assert(order_status_read(order_hash).is_none(), orderbook_errors::ORDER_ALREADY_EXISTS);
-
             match order_type {
                 OrderType::Listing => {
-                    self._create_listing_order(order, order_type, order_hash, user_pubkey);
+                    self._create_listing_order(order, order_type, order_hash);
                 },
                 OrderType::Auction => { self._create_auction(order, order_type, order_hash); },
                 OrderType::Offer => { self._create_offer(order, order_type, order_hash); },
                 OrderType::CollectionOffer => {
                     self._create_collection_offer(order, order_type, order_hash);
                 },
             }
+            self.order_signers.write(order_hash, user_pubkey);
```

Comment:
We should keep the signing info and not just the public key. Because we loose the signature doing that. So we don't have a register of the actual proof that the user has signed the order.

Reply:
@remiroyc actually did that code i just moved it here like we discussed but it's true we should save the whole thing, but maybe @remiroyc add an other idea there

Reply:
we did some modification about it tell me what you think @glihm